src/components/PostsList/PostsList.jsx

```diff
@@ -1,37 +1,56 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
 import './PostsList.scss';
 
-export const PostsList = () => (
-  <div className="PostsList">
-    <h2>Posts:</h2>
-
-    <ul className="PostsList__list">
-      <li className="PostsList__item">
-        <div>
-          <b>[User #1]: </b>
-          sunt aut facere repellat provident occaecati excepturi optio
-        </div>
-        <button
-          type="button"
-          className="PostsList__button button"
-        >
-          Close
-        </button>
-      </li>
-
-      <li className="PostsList__item">
-        <div>
-          <b>[User #2]: </b>
-          et ea vero quia laudantium autem
-        </div>
-
-        <button
-          type="button"
-          className="PostsList__button button"
-        >
-          Open
-        </button>
-      </li>
-    </ul>
-  </div>
-);
+import { getUserPosts } from '../../api/posts';
+
+export const PostsList = ({ selectedUser, selectedPostId, handlePost }) => {
+  const [posts, setPosts] = useState([]);
+
+  useEffect(() => {
+    const getPosts = async() => {
+      const postsFromServer = await getUserPosts(selectedUser);
+
+      setPosts(postsFromServer);
+    };
+
+    getPosts();
+  }, [selectedUser]);
```

Comment:
Let's do it on the upper level, and pass here already prepared posts from App.js